ouroboros-consensus/src/Ouroboros/Consensus/ChainSyncClient.hs

```diff
@@ -474,6 +501,12 @@ chainSyncClient tracer cfg btime (ClockSkew maxSkew)
     disconnect :: ChainSyncClientException blk -> STM m a
     disconnect = throwM
 
+    -- | Trace any 'ChainSyncClientException' if thrown.
+    traceException :: m a -> m a
```

Comment:
Wonder if this merits living in `Util`?

Reply:
Then the type of the exception would have to be passed via a `Proxy`. I propose that we do this when we want to do this at some other place in the code.

Reply:
At that point we might not remember, but sure, not a big deal either way :)